library/core/src/ptr/const_ptr.rs

```diff
@@ -13,6 +13,11 @@ impl<T: ?Sized> *const T {
     /// Therefore, two pointers that are null may still not compare equal to
     /// each other.
     ///
+    /// Note that during const eval, this function may return `false` for out
+    /// of bounds pointers that are in fact `null`. At runtime it is possible
+    /// to offset a pointer in a loop until it becomes a null pointer. This
+    /// will not work in const eval and instead end up in an infinite loop.
```

Comment:
At runtime you can also do something like
```rust
(ptr as *const u8).wrapping_offset(-(ptr as usize))
```
which always returns NULL.

Reply:
sure, but I meant an operation that is actually problematic at compile-time. This operation would return `0`, too, as we'd subtract two abstract pointers to the same alloc.

Reply:
The comment is still strange I think. You can do the offsetting at CTFE time. You just cannot test for NUL.

Reply:
I'd say something like
```
    /// ## Behavior during const evaluation
    ///
    /// When this function is used during const evaluation, it may return `false` for pointers
    /// that turn out to be null at runtime. Specifically, when a pointer to some memory
    /// is offset beyond its bounds in such a way that the resulting pointer is null,
    /// the function will still return `false`. There is no way for CTFE to know
    /// the absolute position of that memory, so we cannot tell if the pointer is
    /// null or not.
```